src/parquet/column_writer.cc

```diff
@@ -294,6 +295,8 @@ ColumnWriter::ColumnWriter(ColumnChunkMetaDataBuilder* metadata,
       num_buffered_encoded_values_(0),
       rows_written_(0),
       total_bytes_written_(0),
+      current_compressed_bytes_(0),
+      flush_on_close_(flush_on_close),
```

Comment:
Would it be better to make this part of `WriterProperties`?

Reply:
It will depend on what RowGroupWriter API we provide to the users. If parquet-cpp takes the responsibility of bounding RowGroup sizes (which I think is not in its scope based on the example above), then this can be.